appcli/variables_manager.py

```diff
@@ -43,86 +55,198 @@ def get_variable(self, path):
         """Gets a value from the configuration.
 
         Args:
-            path (str): Dot notation for the setting. E.g. insilico.external.database.host
+            path (str): Dot notation for the setting. E.g. settings.insilico.external.database.host
 
         Throws:
             Exception: Failed to find the configuration key.
         """
-        configuration = self.__get_configuration()
+        configuration = self._get_configuration()
         try:
             variable = reduce(lambda e, k: e[k], path.split("."), configuration)
         except KeyError as exc:
             raise KeyError(f"Setting [{path}] not set in configuration.") from exc
         return variable
 
     def get_all_variables(self):
-        return self.__get_configuration()
+        return self._get_configuration()
 
     def set_variable(self, path: str, value: Union[str, bool, int, float]):
-        """Sets a value in the configuration.
+        """Sets a value in the configuration
 
         Type of value is not enforced but this might not serialise into yml in a deserialisable format.
 
         Args:
-            path (str): Dot notation for the setting. E.g. insilico.external.database.host
+            path (str): Dot notation for the setting. E.g. settings.insilico.external.database.host
             value: value for the setting
-        """
-        configuration = self.__get_configuration()
-
-        path_elements = path.split(".")
-        parent_path = path_elements[:-1]
-
-        # ensure parent path exists
-        def ensure_path(parent, child):
-            if child not in parent:
-                parent[child] = {}
-            return parent[child]
-
-        reduce(ensure_path, parent_path, configuration)
```

Comment:
From our previous conversation, you said:

> With the original code it could store a value in a non existent field, but if the parent also didn't exist it would throw a NoneType exception.
> This code allows it to create a field at any depth where the parent(s) don't exist.

Isn't this code right here creating the parents?

Reply:
![image](https://user-images.githubusercontent.com/26250869/152286493-4eb84e42-d8ef-4a7a-98bc-f238c3e9be74.png)